service/history/commandChecker.go

```diff
@@ -557,6 +558,18 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
 		return serviceerror.NewInvalidArgument("WorkflowType exceeds length limit.")
 	}
 
+	if timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()) < 0 {
+		return serviceerror.NewInvalidArgument("Invalid WorkflowExecutionTimeout.")
+	}
+
+	if timestamp.DurationValue(attributes.GetWorkflowRunTimeout()) < 0 {
+		return serviceerror.NewInvalidArgument("Invalid WorkflowRunTimeout.")
+	}
+
+	if timestamp.DurationValue(attributes.GetWorkflowTaskTimeout()) < 0 {
```

Comment:
workflow task timeout should never be 0.  We need to make sure it defaults to the server default if one is not specified.

Reply:
this is just verification in case someone set the timeout to be < 0, similar to start workflow.

code below will set the default